### DISCUSSION TITLE
Inter-process swift

### DISCUSSION THREAD

**Discussion by kangz12345** [snu-quiqcl/qiwis#123] [Ideas]
I came up with a new idea: a way to communicate among different processes and adjusting the window positions and dimensions.
1. Wrap each app with a threaded server.
2. Execute each app on a distinct process.
3. A central app (swift?) acts as a message router and a command center.
4. They are connected via sockets.

Then, the server can listen to the messages from swift, and it can change the window position or dimension as ordered.
In that way, swift can set up the windows arrangement.
In this case, we can allow the apps to freely crash and blocks. _Not our business_!

**Comment by kangz12345** [comment]
If we want to implement this, we have to do:
1. Determine how to arrange layout of windows easily.
2. Implement the new swift again from the beginning...

@BECATRUE @Ahrrri

**Reply by BECATRUE** [reply]
Wow.. It looks so awesome!

Like your idea, communication with only socket will run errorless.
Then, should we start each app individually?

**Reply by kangz12345** [reply]
Yes, or alternatively it can be forked from swift.
